### PR TITLE
use formatToParts

### LINKED ISSUE #2: Fails in Chrome
`Error: Pseudo-ISO failed, got: 18/02/2020, 13:00:00`

Apparently Chrome's implementation of `Intl.DateTimeFormat` inserts a comma into the string.

Nice lib by the way!

### PR DESCRIPTION
**Pull request by gombosg**
Trying to fix #2 - what do you think?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -38,11 +37,10 @@ function getPseudoISO (date, timeZone) {
...
     hour12: false,
     timeZone
-  }, 'af-ZA', timeZone).format(date)
-  if (str.length !== 19 || str[10] !== ' ') {
-    throw new Error('Pseudo-ISO failed, got: ' + str)
-  }
-  return str.replace(' ', 'T') + 'Z'
+  }, 'en-US', timeZone).formatToParts(date)
+  const find = (key) => formattedParts.find(part => part.type === key).value
+  const utcDate = `${find('year')}-${find('month')}-${find('day')}T${find('hour')}:${find('minute')}:${find('second')}Z`
```
Comment (gregolai) [inline index.js]: All these `find`s are performance-intensive. You can refactor this to loop through `formattedParts` only once, extracting year/month/day/etc

File: index.js [hunk 2]
```diff
@@ -38,11 +37,10 @@ function getPseudoISO (date, timeZone) {
...
+  return new Date(utcDate)
```
Comment (gregolai) [inline index.js]: It used to return a string, but now it returns a `Date` instance?

File: index.js (hunk 1 above)
Comment (gombosg) [reply index.js]: Got it, switched to a `switch`.

File: index.js (hunk 2 above)
Comment (gombosg) [reply index.js]: True, oops! The error wasn't apparent since it got `new Date`'d in `getOffset`.

**Comment by gombosg** [comment]
Sorry for the late reply, this got lost in my Github feed.
I pushed some fixes, let me know how you like it. I actually just copied these few lines into our production code, but it works great so far.

**Pull request closed without merging**